### PR TITLE
Adding an example dag for dynamic task mapping

### PR DESCRIPTION
**Pull request by amoghrajesh**
As per the issue #26677, it would be helpful to have a dynamic task mapping dag example in airflow dag examples so that users can try it directly.

Building on top of the closed PR: https://github.com/apache/airflow/pull/26678, raising a PR to fix this issue.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: docs/apache-airflow/concepts/dynamic-task-mapping.rst [hunk 1]
```diff
@@ -30,27 +30,7 @@ Simple mapping
...
-            return x + 1
-
-        @task
-        def sum_it(values):
-            total = sum(values)
-            print(f"Total was {total}")
-
-        added_values = add_one.expand(x=[1, 2, 3])
-        sum_it(added_values)
+If you want to see a simple usage of Dynamic Task Mapping, you can look at the ``example_dynamic_task_mapping.py`` example DAG that comes with Airflow.
```
Comment (uranusjr) [inline docs/apache-airflow/concepts/dynamic-task-mapping.rst]: You can use `exampleinclude` to include the code here. Search for this keyword for examples.

Comment (amoghrajesh) [reply docs/apache-airflow/concepts/dynamic-task-mapping.rst]: Made the required changes. Can you have a look again?

**Comment by amoghrajesh** [comment]
@uranusjr working on fixing the build failures. Might need some help around it

**Review by uranusjr — APPROVED** [review]

**Comment by amoghrajesh** [comment]
@uranusjr the build seems to fail. I need some guidance on what's failing

**Comment by amoghrajesh** [comment]
@potiuk @uranusjr I am running into this doc build error:
```
############################## Start docs build errors summary ##############################

============================== apache-airflow ==============================
------------------------------ Error   1 --------------------
 ERROR: Unexpected indentation.

File path: apache-airflow/concepts/dynamic-task-mapping.rst (35)

  30 | 
  31 | In its simplest form you can map over a list defined directly in your DAG file using the ``expand()`` function instead of calling your task directly.
  32 | 
  33 | If you want to see a simple usage of Dynamic Task Mapping, you can look below:
  34 | .. exampleinclude:: /../../airflow/example_dags/example_dynamic_task_mapping.py
> 35 |     :language: python
  36 |     :dedent: 4
  37 | 
  38 | This will show ``Total was 9`` in the task logs when executed.
  39 | 
  40 | This is the resulting DAG structure:

```

I am not really able to figure out whats wrong here. The indent looks about right to me. Can i get some hint?

**Comment by amoghrajesh** [comment]
@potiuk @uranusjr The UTs are failing in the `validate_deserialized_task`, I am not sure how to debug it. Can someone guide me here? https://github.com/apache/airflow/actions/runs/3692197454/jobs/6251170704

**Comment by potiuk** [comment]
First of all - rebase to latest `main`. You are 20 commits behind, and likely it is coming from a broken commit of someone else. You should lways rebase first if you see something strange. You will find guide about rebasing in our CONTRIBUTING.rst docs.

image

If it does not help, I guess this is some a side effect - likely because you are using the same dag_id/task_id or smth that clashes between them.  look at similarities between your tests and the one that is failing. The way you reproduce it and test is just to run the failing test - ideally in the Breeze environment where the test is running in the same environment as in CI. The usual debugging techniques - logging and debugger are helpful.

Unfortunately, our tests are not as isolated as I would like them to be. But I might have some  time soon (and I alredy have an idea how to increase an isolation to avoid such side-effects in the future), but for now I think just try to reproduce it locally.

**Comment by amoghrajesh** [comment]
Thanks for the pointers @potiuk 
Trying a rebase. If it does not fix it, I will try running it locally and fixing it!

**Comment by uranusjr** [comment]
Ah, this actually breaks a couple tests that assume all tasks loaded from example DAGs are BaseOperator, haha. I’ll fix those for you.

**Comment by amoghrajesh** [comment]
Haha okay @uranusjr 
We will have to wait on merging this PR i guess then?

**Comment by uranusjr** [comment]
I pushed a fix to the tests. Hopefully things should work now.

**Comment by amoghrajesh** [comment]
@uranusjr looks like the previous build was "cancelled". Rebasing the PR again.

**Comment by amoghrajesh** [comment]
@potiuk @uranusjr Finally we have a green build! Can you help in merging this issue?

**Comment by potiuk** [comment]
@amoghrajesh with uttermost pleasure. Thanks @uranusjr for fixing those "assumptions" in example tests :)

**Pull request merged**